src/Aop/Pointcut/PointcutParser.php

```diff
@@ -25,4 +26,12 @@ public function __construct(PointcutGrammar $grammar)
         $parseTable = include 'PointcutParseTable.php';
         parent::__construct($grammar, $parseTable);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parse(TokenStream $stream)
+    {
+        return new AndPointcut(parent::parse($stream), new ClassDeclaresPointcut());
```

Comment:
Ok, so this is why I think that this solution is kinda hackish...

Whatever our parser gets as a pointcut filter, it has to be additionally filtered. So it has to be applied last.

`parent::parse($stream)` -> will provide a pointcut that will match something, and that something can be a collection of properties and methods, right?

So, whatever methods and properties we got, we have to apply `ClassDeclaresPointcut` matcher to filter out non-declared properties and methods.

I guess that Grammar should give us this expression, is that even possible?